packages/typespec-autorest/src/openapi.ts

```diff
@@ -1482,12 +1470,8 @@ export async function getOpenAPIForService(
   ): OpenAPI2HeaderParameter {
     const base = getOpenAPI2ParameterBase(prop, name);
     const headerOptions = getHeaderFieldOptions(program, prop);
-    // eslint-disable-next-line @typescript-eslint/no-deprecated
-    let collectionFormat = headerOptions.format;
-    if (
-      !collectionFormat &&
-      (typeof headerOptions.explode === "boolean" || $.array.is(prop.type))
-    ) {
+    let collectionFormat;
+    if (typeof headerOptions.explode === "boolean" || $.array.is(prop.type)) {
       collectionFormat = headerOptions.explode ? "multi" : "csv";
     }
     if (collectionFormat && !["csv", "ssv", "tsv", "pipes"].includes(collectionFormat)) {
```

Comment:
I think this technically cann't happen now? should the error also be removed?

Might technically have to check for the encode for `ssv` and `pipes` though I don't think anyone is using it in spec repo so we could not support them for now

Reply:
Lets leave it for now as it won't get hit. If TCGC is updated with consolidated `@encode` logic, we can added it back top and this check could be live again.

Reply:
i've added `ssv` and `pipes` support for tcgc and autorest emitter.